package messageTypeEnum;

message typeEnums{
	enum mapping {
		HEART_BEAT = 1;
		HEART_BEAT_ACK = 2;
		OSD_START_MONITORING = 3;
		OSD_START_MONITORING_ACK = 4;
		RECV_PROC_START_MONITORING = 5;
		RECV_PROC_START_MONITORING_ACK = 6;
		LL_START_MONITORING = 7;
		LL_START_MONITORING_ACK = 8;
		GET_GLOBAL_MAP = 9;
		GLOBAL_MAP_INFO = 10;
		COMP_TRANSFER_INFO = 11;
		COMP_TRANSFER_INFO_ACK = 12;
		COMP_TRANSFER_FINAL_STAT = 13;
		COMP_TRANSFER_FINAL_STAT_ACK = 14;
		GET_SERVICE_COMPONENT = 15;
		GET_SERVICE_COMPONENT_ACK = 16;
		TRANSFER_COMPONENT = 17;
		TRANSFER_COMPONENT_RESPONSE = 18;
		GET_OBJECT_VERSION = 19;
		GET_OBJECT_VERSION_ACK = 20;
		NODE_ADDITION_CLI = 50;
		NODE_ADDITION_CLI_ACK = 51;
		NODE_ADDITION_GL = 52;
		NODE_ADDITION_GL_ACK = 53;
		NODE_RETIRE = 54;
		NODE_RETIRE_ACK = 55;
		NODE_SYSTEM_STOP_CLI = 56;
		LOCAL_NODE_STATUS = 57;
		LOCAL_NODE_STATUS_ACK = 58;
		NODE_STATUS = 59;
		NODE_STATUS_ACK = 60;
		NODE_STOP_CLI = 61;
		NODE_STOP_CLI_ACK = 62;
		STOP_SERVICES = 63;
		STOP_SERVICES_ACK = 64;
		NODE_FAILOVER = 65;
		NODE_FAILOVER_ACK = 66;
		TAKE_GL_OWNERSHIP = 67;
		TAKE_GL_OWNERSHIP_ACK = 68;
		BLOCK_NEW_REQUESTS = 69;
		BLOCK_NEW_REQUESTS_ACK = 70;
		NODE_DELETION = 71;
		NODE_DELETION_ACK = 72;
		NODE_REJOIN_AFTER_RECOVERY = 73;
		NODE_REJOIN_AFTER_RECOVERY_ACK = 74;
		GET_CLUSTER_STATUS = 75;
		GET_CLUSTER_STATUS_ACK = 76;
		NODE_STOP_LL = 77;
		NODE_STOP_LL_ACK = 78;
		CONTAINER_UPDATE = 79;
		RELEASE_TRANSACTION_LOCK = 80;
		STATUS_ACK = 81;
		NODE_ADDITION_FINAL_ACK = 82;
		INITIATE_CLUSTER_RECOVERY = 83;
		ERROR_MSG = 255;
	}

	optional mapping type = 1;
}
